### ISSUE TITLE
MongoDB: WriteConcern not used when updating an entity.

### ISSUE THREAD

**Issue by vreniers** [Impetus/kundera#830]
Hi,

I was debugging my code in order to validate that the WriteConcern I set was actually being used.
I've set WriteConcern using the following method:

``` java
EntityManager em = emf.createEntityManager();
em.setProperty(MongoDBClientProperties.WRITE_CONCERN, WriteConcern.ACKNOWLEDGED);
```

Which works just fine and WriteConcern.ACKNOWLEDGED is used for inserting records.
However, on update I noticed this was not the case.

``` java
User user = new User();
user.setUserId(key);
user.setField0("Some value");

em.merge(user);
```

I noticed when debugging that Kundera does not pass on the WriteConcern when saving the updated object. I traced this back to the following [line of code](https://github.com/impetus-opensource/Kundera/blob/trunk/src/kundera-mongo/src/main/java/com/impetus/client/mongodb/MongoDBClient.java#L1146).
A simple fix involves calling the other save method and supplying the write concern.

**Comment by chhavigangwal** [comment]
@vreniers 

Thanks for pointing out. We'll verify it at our end and fix it if required.

Chhavi

**Comment by devender-yadav** [comment]
@vreniers 

This is a bug. We will fix it in the next release. Yes, this will be fixed by calling 

```
  WriteResult save(DBObject jo,  WriteConcern concern)
```

 By the way,  default writeconcern property for the Kundera-MongoDB is **ACKNOWLEDGED**.

-Dev

**Comment by devender-yadav** [comment]
@vreniers 

I added a fix in the trunk. This fix will be available from next release (**3.4**). Meanwhile, you can build Kundera from trunk use `3.4-SNAPSHOT` version if needed. 

-Dev

**Comment by vreniers** [comment]
Thank you.
